### DISCUSSION TITLE
diesel_cli compiled features flag

### DISCUSSION THREAD

**Discussion by alexxroche** [diesel-rs/diesel#2717] [Ideas]
Could we add to the version output which features have been compiled into diesel_cli ? I spent two days trying to get diesel to work with postgres, (when it would work with sqlite) and found out that cargo had been failing to re-compile with postgres support.

Ideally:

```bash
diesel -V
diesel 1.3.2 "postgres mysql"

```
or
```bash
diesel --version
diesel 1.4.1 "postgres sqlite"
```

If we need to maintain 100% backwards compatibility then add a new flag 
```bash
diesel -h
FLAGS:  
    -f, --features
            List which features have been compiled into this version of diesel_cli

diesel -f
postgres sqlite mysql
```

**Comment by weiznich** [comment]
This sounds like a great idea. Such a contribution would definitively be welcome.

**Reply by alexxroche** [reply]
I've sketched out an this idea: https://github.com/diesel-rs/diesel/pull/2724
I picked the "maintain backward compatibility", (though obviously the developers of diesel can implement this how they see best.)
Still needs unit_tests for this new feature to ensure that: 
`cargo build --package diesel_cli --no-default-features --features "sqlite postgres mysql"` still works,
and an integration test for `../target/debug/diesel features`.
